index.html

```diff
@@ -89,46 +143,283 @@
+          .attr("dy", ".71em")
+          .style("text-anchor", "middle")
+          .text("Math proficiency");
+   
+
+
+        /*     
+          .append("text")
+          .attr("transform", "rotate(-90)")
+          .attr("y", 0 - margin.left)
+          .attr("dy", ".7em")
+          .style("text-anchor", "end")
+          .text("Math proficiency");
+*/
+
+    
     // EVENT HANDLERS
 
-    d3.select('#year-range').on('change', function () {
-      options.yearRange = d3.event.target.value;
+    d3.select('#categories').on('change', function () {
+      options.categoryRange = d3.event.target.value;
       update();
     });
   }
 
   function update() {
     // TRANSFORM
-    txData = data.slice();
+    schoolData = data.slice();
+/*
+    data.forEach(function (d) { 
+      d.filtered = d.Category;
+      var schoolData = data.filter(function (d) {return d.filtered;});
+    })
+*/
 
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
-      txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
+    if (options.categoryRange !== 'all') {
+     // var categoryRange = options.categoryRange.split('');
+      schoolData = schoolData.filter(function (d) {
+      //  return d.Category >= categoryRange[0] && d.Category <= categoryRange[1];
+          return d.Category === d3.event.target.value;
       });
+
     }
 
     // SCATTERPLOT
 
+   // svg.selectAll('.point')
+    //  .remove();
+
+  var points = svg.selectAll('.point') //point is used instead of circles, in case different shapes are used
+    .data(schoolData);
+
+    //console.log(points)
+
+    points.enter().append('circle')  //add points if points don't exist
+      .attr('class', 'point')
+      .attr('r', 5)
+
+    points    //all points
+      .attr('cx', function (d) { return x(d.Reading_proficient/100); })
+      .attr('cy', function (d) { return y(d.Math_proficient/100); });
+
+   points.exit().remove();
+
     // HISTOGRAMS
 
+    //X-AXIS
+
+    var hist_x = d3.layout.histogram()
+          .value(function(d) { return d.Reading_proficient/100; })
+          .range([0, d3.max(schoolData, function(d){ return d.Reading_proficient/100; }) ])
+          .bins(x.ticks(10))
+
+    var layout = hist_x(schoolData);
+
+    for(var i = 0; i < layout.length; i++) {
+    var bin = layout[i];
+    var colorScale = d3.scale.category20();
+    svg.selectAll("circle")
+    .data(bin, function(d) { return d.Reading_proficient/100; })
+    .style("fill", function() {  return colorScale(i); })
+
+     }
+
+    var maxLength = d3.max(layout, function(d) { return d.length });
+    var heightScale = d3.scale.linear()
+    .domain([0, maxLength])
+    .range([0, height/5]);
+
+    var xScale = d3.scale.ordinal()
+    .domain(d3.range(10))
+    .rangeBands([0, width], 0.2)
+
+ //   .bins(x.ticks(10));
+
+   var rectsX = svg.selectAll("rect")
+      .data(layout)
+      .enter()
```

Comment:
This is the problem here. Note how when we create `points` starting on line 308, we stop after the `.data()` call. Then we call `points.enter()` on line 313. But here, we're calling `.enter()` immediately, which means that its result is getting assigned to `rectsX`, and its result is just the newly created rects. This means there is no `.exit()` group to call, and it also means that this won't ever get updated after the first time through, because all of the rects already exist.